### PR TITLE
Validation folder:

### PR DESCRIPTION
**Pull request by karthikamurthy**
Contains files related to argument validation and json schema validation

Signed-off-by: Karthika Murthy <karthika.murthy@intel.com>

### CODE REVIEW

File: src/internal/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (manju956) [inline src/internal/__init__.py]: Missing copyright and license header

File: src/internal/validation/argument_validator.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+Argument Validation
+""" 
+
+from enums.worker import WorkerType, WorkerStatus
+from exceptions.invalid_parameter import InvalidParamException
+
+
+class ArgumentValidator(object):
+    """
+    Helper method for validating an argument that will be used by this API in any requests.
```
Comment (manju956) [inline src/internal/validation/argument_validator.py]: reword Helper method to Helper class

File: src/internal/validation/data/WorkOrderReceiptCreate.json [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        "receiptCreateStatus": {
+            "type": "integer",
+            "error_msg":
+                "Invalid data format for receiptCreateStatus"
+            },
+        "workOrderRequestHash": {
+            "type": "string",
+            "contentEncoding": "base64",
+            "error_msg":
+                "Invalid data format for Worker id"
```
Comment (manju956) [inline src/internal/validation/data/WorkOrderReceiptCreate.json]: Invalid data format for workOrderRequestHash

File: src/internal/validation/data/WorkOrderReceiptCreate.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            },
+        "signatureRules": {
+            "type": "string",
+            "error_msg":
+                "Invalid data format for signatureRules"
+            },
+        "receiptVerificationKey": {
+            "type": "string",
+            "error_msg":
+                "Invalid data format for signatureRules"
```
Comment (manju956) [inline src/internal/validation/data/WorkOrderReceiptCreate.json]: Invalid data format for receiptVerificationKey

File: src/internal/validation/data/WorkOrderSubmit.json [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+        }
+    },
+    "outData": {
+        "type": "array",
+        "uniqueItems": true,
+        "additionalProperties": false,
+        "items": {
+        "type": "object",
+        "error_msg":
+        "Invalid data format for outData ",
```
Comment (manju956) [inline src/internal/validation/data/WorkOrderSubmit.json]: remove extra space at the end of string

File: src/internal/validation/data/WorkerLookUp.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
+{
+    "type": "object",
+    "uniqueItems": true,
+    "additionalProperties": false,
+    "properties": {
+        "workerType": {
+            "type": "integer",
+            "enum": [1, 2, 3],
+            "error_msg":
+                "WorkType should be an Integer of range 1-3"},
```
Comment (manju956) [inline src/internal/validation/data/WorkerLookUp.json]: correct typo workType -> workerType

File: src/internal/validation/data/WorkerLookUpNext.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
+{
+    "type": "object",
+    "uniqueItems": true,
+    "additionalProperties": false,
+    "properties": {
+        "workerType": {
+            "type": "integer",
+            "enum": [1, 2, 3],
+            "error_msg":
+                "WorkType should be an Integer of range 1-3"},
```
Comment (manju956) [inline src/internal/validation/data/WorkerLookUpNext.json]: correct typo workType -> workerType

File: src/internal/validation/proxy/argument_validator.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+def not_valid_contract(self, **kwargs): 
+    for key, value in kwargs.items():
+        if value is None:
+            message = key.replace("_", " ") + " is not avaliable !!"
+            raise ContractFailed(message)
+    return True
+
+setattr(ArgumentValidator,'not_valid_contract', not_valid_contract)
+
```
Comment (manju956) [inline src/internal/validation/proxy/argument_validator.py:33]: remove additional empty lines at the end of the file

**Review by manju956 — CHANGES_REQUESTED** [review]

File: src/internal/validation/argument_validator.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+            "WorkerType" : WorkerType,
+            "WorkerStatus" : WorkerStatus
+        }
+      if ArgumentValidator.__instance != None:
+         raise Exception("This class is a singleton!")
+      else:
+        ArgumentValidator.__instance = self
+
+
+    def enum_value(self,id, enum, value):
```
Comment (manojgop) [inline src/internal/validation/argument_validator.py:49]: function, parameter, return value description missing in this and other functions in this file.

File: src/internal/validation/proxy/argument_validator.py [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+# limitations under the License.
+
+"""
+Argument Validation Extension
+""" 
+
+from validation.argument_validator import ArgumentValidator
+from exceptions.proxy.contract_error import ContractFailed
+
+def not_valid_contract(self, **kwargs):
```
Comment (manojgop) [inline src/internal/validation/proxy/argument_validator.py]: Can we make this function as part of ProxyArgValidator class which can be derived from Base ArgumentValidator class

File: src/internal/validation/json_validator.py [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+# limitations under the License.
+
+
+import json
+import pkg_resources
+from jsonschema import validate, ValidationError, SchemaError
+from exceptions.invalid_parameter import InvalidParamException
+
+
+def json_validation(id, method, params):
```
Comment (manojgop) [inline src/internal/validation/json_validator.py]: Define a class and make these as member function of the class

**Review by manojgop — CHANGES_REQUESTED** [review]

**Review by manju956 — APPROVED** [review]

File: src/internal/validation/argument_validator.py (hunk 9 above)
Comment (karthikamurthy) [reply src/internal/validation/argument_validator.py:49]: done

File: src/internal/validation/proxy/argument_validator.py (hunk 8 above)
Comment (karthikamurthy) [reply src/internal/validation/proxy/argument_validator.py:33]: done

**Review by manojgop — APPROVED** [review]

**Pull request merged**